skyportal/models.py

```diff
@@ -2786,16 +2831,49 @@ def from_ascii(
 #        return '/' + file_uri.lstrip('./')
 
 
+def updatable_by_token_with_listener_acl(cls, user_or_token):
+    if user_or_token.is_admin:
+        return public.query_accessible_rows(cls, user_or_token)
+
+    instruments_with_apis = (
+        Instrument.query_records_accessible_by(user_or_token)
+        .filter(Instrument.listener_classname.isnot(None))
+        .all()
+    )
+
+    api_map = {
+        instrument.id: instrument.listener_class.get_acl_id()
+        for instrument in instruments_with_apis
+    }
+
+    accessible_instrument_ids = [
+        instrument_id
+        for instrument_id, acl_id in api_map.items()
+        if acl_id in user_or_token.permissions
+    ]
+
+    return (
+        DBSession()
+        .query(cls)
+        .join(Allocation)
+        .join(Instrument)
+        .filter(Instrument.id.in_(accessible_instrument_ids))
+    )
+
+
 class FollowupRequest(Base):
     """A request for follow-up data (spectroscopy, photometry, or both) using a
     robotic instrument."""
 
     # TODO: Make read-accessible via target groups
     create = read = AccessibleIfRelatedRowsAreAccessible(obj="read", allocation="read")
     update = delete = (
-        AccessibleIfUserMatches('allocation.group.users')
-        | AccessibleIfUserMatches('requester')
-    ) & read
+        (
+            AccessibleIfUserMatches('allocation.group.users')
+            | AccessibleIfUserMatches('requester')
+        )
+        & read
+    ) | CustomUserAccessControl(updatable_by_token_with_listener_acl)
```

Comment:
This allows remote facilities to update, but not read, any followup request associated with an instrument that the remote facility implements an API for. Requesters or others in the allocation group can also update the request.